mangaki/mangaki/views.py

```diff
@@ -807,6 +809,120 @@ def legal_mentions(request):
+    except PageNotAnInteger:
+        suggestions = paginator.page(1)
+    except EmptyPage:
+        suggestions = paginator.page(paginator.num_pages)
+
+    context = {
+        'suggestions': suggestions
+    }
+
+    return render(request, 'fix/fix_index.html', context)
+
+
+def fix_suggestion(request, suggestion_id):
+    cluster_colors = {
+        'unprocessed': 'text-info',
+        'accepted': 'text-success',
+        'rejected': 'text-danger'
+    }
+
+    # Retrieve the Suggestion object if it exists else raise a 404 error
+    suggestion = get_object_or_404(
+        Suggestion.objects.select_related('work', 'user', 'work__category').annotate(
+            count_agrees=Count(Case(When(evidence__agrees=True, then=1))),
+            count_disagrees=Count(Case(When(evidence__agrees=False, then=1)))
+        ),
+        id=suggestion_id
+    )
+
+    # Retrieve the Evidence object if it exists
+    evidence = None
+    if request.user.is_authenticated:
+        try:
+            evidence = Evidence.objects.get(user=request.user, suggestion=suggestion)
+        except ObjectDoesNotExist:
+            evidence = None
+
+    # Retrieve related clusters
+    clusters = WorkCluster.objects.select_related(
+        'resulting_work', 'resulting_work__category').prefetch_related(
+        'works', 'works__category', 'checker').filter(origin=suggestion_id).all()
+    colors = [cluster_colors[cluster.status] for cluster in clusters]
+
+    # Get the previous suggestion, ie. more recent and of the same checked status
+    previous_suggestions_ids = Suggestion.objects.filter(date__gt=suggestion.date,
+        is_checked=suggestion.is_checked).order_by('date').values_list('id', flat=True)
+
+    # If there is no more recent suggestion, and was checked, just pick from not checked suggestions
+    if not previous_suggestions_ids and suggestion.is_checked:
+        previous_suggestions_ids = Suggestion.objects.filter(is_checked=False).order_by('date').values_list('id', flat=True)
+
+    # Get the next suggestion, ie. less recent and of the same checked status
+    next_suggestions_ids = Suggestion.objects.filter(date__lt=suggestion.date,
+        is_checked=suggestion.is_checked).order_by('-date').values_list('id', flat=True)
+
+    # If there is no less recent suggestion, and wasn't checked, just pick from checked suggestions
+    if not next_suggestions_ids and not suggestion.is_checked:
+        next_suggestions_ids = Suggestion.objects.filter(is_checked=True).order_by('-date').values_list('id', flat=True)
+
+    context = {
+        'suggestion': suggestion,
+        'clusters': zip(clusters, colors) if clusters and colors else None,
+        'evidence': evidence,
+        'next_id': next_suggestions_ids[0] if next_suggestions_ids else None,
+        'previous_id': previous_suggestions_ids[0] if previous_suggestions_ids else None
+    }
+
+    return render(request, 'fix/fix_suggestion.html', context)
+
+
+def update_evidence(request):
```

Comment:
I think we should add [`@login_required`](https://docs.djangoproject.com/fr/1.11/topics/auth/default/#django.contrib.auth.decorators.login_required) here, in fact, we should not be able to call this method if we are not logged in.